### PR TITLE
sys-libs/ncurses: Remove libcurses.so symlink

### PR DESCRIPTION
**Pull request by mgorny**
Closes: https://bugs.gentoo.org/836696
Signed-off-by: Michał Górny <mgorny@gentoo.org>

### CODE REVIEW

File: sys-libs/ncurses/ncurses-6.3_p20211106-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+	# Move main libraries into /.
+	if multilib_is_native_abi ; then
+		gen_usr_ldscript -a \
+			"${NCURSES_TARGETS[@]}" \
+			$(usex tinfo 'tinfow tinfo' '')
+	fi
+	if ! tc-is-static-only ; then
+		# Provide a link for -lcurses.
+		ln -sf libncurses$(get_libname) "${ED}"/usr/$(get_libdir)/libcurses$(get_libname) || die
+	fi
```
Comment (Polynomial-C) [inline sys-libs/ncurses/ncurses-6.3_p20211106-r1.ebuild:286]: So here the `libcurses*` symlink gets created...

File: sys-libs/ncurses/ncurses-6.3_p20211106-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+	# -FIXME-
+	# Ugly hackaround for riscv having two parts libdir (#689240)
+	# Replace this hack with an official solution once we have one...
+	# -FIXME-
+	dosym $(sed 's@[^/]\+@..@g' <<< $(get_libdir))/share/terminfo \
+		/usr/$(get_libdir)/terminfo
+
+	# Remove obsolete libcurses symlinks
+	# https://bugs.gentoo.org/836696
+	rm "${ED}"/usr/$(get_libdir)/libcurses* || die
```
Comment (Polynomial-C) [inline sys-libs/ncurses/ncurses-6.3_p20211106-r1.ebuild:302]: ... and here it gets removed again? Interesting logic...

Comment (mgorny) [reply sys-libs/ncurses/ncurses-6.3_p20211106-r1.ebuild:302]: Thank you for your respectful comment. Unfortunately, it seems that you are only partially right and the removal needs to stay. I'm sorry that your sarcasm backfires.

**Comment by floppym** [comment]
It looks like it might be "cleaner" to disable this via the build system by passing `--disable-overwrite` to configure.

**Comment by mgorny** [comment]
> It looks like it might be "cleaner" to disable this via the build system by passing `--disable-overwrite` to configure.

This causes "baseline" headers to be moved into `ncurses` subdirectory. Is that desirable?

**Comment by floppym** [comment]
> This causes "baseline" headers to be moved into `ncurses` subdirectory. Is that desirable?

Yeah, I realized that after looking at it further. I think this will probably cause some additional build-time failures for packages that don't use pkg-config properly, but fixing these might be better in the long run.

**Comment by floppym** [comment]
Let's stick with just removing the libcurses symlinks for now, and tackle the headers relocation separately, at a later date.

**Review by floppym — APPROVED** [review]

File: sys-libs/ncurses/ncurses-6.3_p20211106-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+
+	# Build fails to create this ...
+	# -FIXME-
+	# Ugly hackaround for riscv having two parts libdir (#689240)
+	# Replace this hack with an official solution once we have one...
+	# -FIXME-
+	dosym $(sed 's@[^/]\+@..@g' <<< $(get_libdir))/share/terminfo \
+		/usr/$(get_libdir)/terminfo
+
+	# Remove obsolete libcurses symlinks
```
[the remark this answers is not in the document]
Comment (mgorny) [reply sys-libs/ncurses/ncurses-6.3_p20211106-r1.ebuild]: Done.

**Pull request closed without merging**